packages/runtime/container-runtime/src/summarizer.ts

```diff
@@ -155,8 +158,14 @@ export class Summarizer extends EventEmitter implements ISummarizer {
      * the run promise, and also close the container.
      * @param reason - reason code for stopping
      */
-    public stop(reason: SummarizerStopReason) {
+    public stop(reason: SummarizerStopReason, forceClose = false) {
```

Comment:
Don't we always want to close the runtime when summarizer is stopped? In what scenario would we not want to do that?

Reply:
No. Normal flow should allow runCore() to run through last summary process.
I.e. if we go to runCore, then it should manage whole flow end-to-end.
The problem here is that we never get to run(), and thus nobody closes container.
There might be other paths were we have same issue but we get away because container losing connection results in closing of container. In this case we likely were not yet connected when check run...

Reply:
I was contemplating to have two methods to have it more explicit. Or not have boolean arg at all and use only internal state (this.running). But I like explicit flows, i.e. be ablet to look at callers and say for sure what happens...

Reply:
Should the code simply say - if (!this.running) and someone is trying to stop, close the container? Rather than relying on the caller to have this information.